lib/headset_event.dart

```diff
@@ -56,16 +56,16 @@ class HeadsetEvent {
   Future<dynamic> _handleMethod(MethodCall call) async {
     switch(call.method) {
       case "connect":
-        return detectPluggedCallback(HeadsetState.CONNECT);
+        return detectPluggedCallback!(HeadsetState.CONNECT);
```

Comment:
Looks like `detectPluggedCallback` can be null here. Maybe it's better to create a local variable and assign it to the value of `detectPluggedCallback`. Then check if that's null, if it is - just return. And if not, then there's the switch, and the local variable can be used instead of `detectPluggedCallback`, with no `!` operator, as we already checked for null.

We use local variable copy and not `detectPluggedCallback` directly, as it's value can be changed while we are still inside of the method. Very low chance, yes, but still possible.

Reply:
All Things Considered by you 👍